lib/libs/preferences.dart

```diff
@@ -240,8 +251,118 @@ class Preferences extends SerializableObject with ChangeNotifier {
     _loggingEnabled = loggingEnabled;
   }
 
+  /// Prefs aus externer Datei laden
+  Future<String> loadFromExportJson(BuildContext context) async {
+    final sie = preferredPronoun == Pronoun.sie;
+    /// Datei-Auswahldialog
+    FilePickerResult? result = await FilePicker.platform.pickFiles();
+    /// Prüft, ob eine Datei ausgewählt wurde
+    if (result != null) {
+      File file = File(result.files.single.path!);
+      try {
+        var importJsonText = (await file.readAsString());
+        var importJson = jsonDecode(importJsonText);
+
+        if (importJson['prefs_version'] != prefsVersion) {
+          bool abort = false;
+          await showDialog(
+            context: context,
+            builder: (context) => AlertDialog(
+              title: const Text("Achtung!"),
+              content: Text("Die Version dieser App stimmt nicht mit der Version der Herkunftsapp der Datei überein. ${sie ? "Bitte aktualisieren Sie": "Bitte aktualisiere"} beide Apps und ${sie ? "versuchen Sie" : "versuche"} es erneut. Das Fortfahren kann zu Fehlern führen und ${sie ? "Sie sollten dies nur benutzen, wenn Sie wissen, was Sie tun!": "Du solltest dies nur benutzen, wenn Du weißt, was du tust!"}"),
+              actions: [
+                TextButton(
+                  onPressed: () => Navigator.pop(context),
+                  child: const Text("Trotzdem fortfahren"),
+                ),
+                TextButton(
+                  onPressed: () {
+                    Navigator.pop(context);
+                    abort = true;
+                  },
+                  child: const Text("Abbrechen"),
+                ),
+              ]
+            ),
+          );
+          if (abort) return "abort";
+        }
+        loadFromJson(importJson["prefs_json"].toString());
+        return "success";
+      } catch (e, s) {
+        logCatch("prefs_import", e, s);
+        return "import_error";
+      }
+    } else {
+      return "abort";
+    }
+  }
+
   Preferences() {
     objectCreators["time_to_next_plan"] = (_) => HMTime(14, 45);
     objectCreators["stuplan_names"] = (_) => <String>[];
   }
 }
+
+Widget Function(BuildContext) sharePreferencesPageBuilder(String? data) =>
+        (context) => SharePreferencesPage(data!);
+class SharePreferencesPage extends StatefulWidget {
+  final String data;
+  const SharePreferencesPage(this.data, {super.key});
+  @override
+  State<SharePreferencesPage> createState() => _SharePreferencesPageState();
+}
+class _SharePreferencesPageState extends State<SharePreferencesPage> {
+  @override
+  Widget build(BuildContext context) {
+    String prefsJson = sharedPreferences.getString(prefsPrefKey) as String;
```

Comment:
Ich würde sehr empfehlen, mit dem Kontext die `Preferences` zu holen, in den `Preferences` die Funktion `_serialize` öffentlich zu machen und dann diese hier aufrufen, statt das aus den shared prefs auszulesen.